tests/kafkatest/services/performance/producer_performance.py

```diff
@@ -169,6 +158,8 @@ def _worker(self, idx, node):
         except:
             raise Exception("Unable to parse aggregate performance statistics on node %d: %s" % (idx, last))
 
+        self.http_metrics_stop()
```

Comment:
It seems a bit asymmetric to have to call `stop` in the subclass when we never had to call start. It might be missed in future subclasses, causing the thread to leak.